src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -243,5 +241,44 @@ private BoundNode CollectDynamicAnalysis(BoundStatement statement)
             
             return statementFactory.Block(ImmutableArray.Create(cellAssignment, statement));
         }
+
+        private static CSharpSyntaxNode SyntaxForSpan(BoundStatement statement)
+        {
+            CSharpSyntaxNode syntaxForSpan;
+
+            switch (statement.Kind)
+            {
+                case BoundKind.IfStatement:
+                    syntaxForSpan = ((BoundIfStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.WhileStatement:
+                    syntaxForSpan = ((BoundWhileStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.ForEachStatement:
+                    syntaxForSpan = ((BoundForEachStatement)statement).Expression.Syntax;
+                    break;
+                case BoundKind.DoStatement:
+                    syntaxForSpan = ((BoundDoStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.UsingStatement:
+                    {
+                        BoundUsingStatement usingStatement = (BoundUsingStatement)statement;
+                        BoundNode operand = (BoundNode)usingStatement.ExpressionOpt ?? usingStatement.DeclarationsOpt;
+                        syntaxForSpan = operand != null ? operand.Syntax : usingStatement.Syntax;
```

Comment:
Is it possible for the operand be null (is it possible to have a using statement that has neither an expression nor declaration(s)?